File: java/src/main/java/todo/TodoService.java
package todo;

import io.vavr.collection.List;
import lombok.AllArgsConstructor;
import lombok.NonNull;

@AllArgsConstructor
public class TodoService {
    private final TodoRepository repository;

    public List<Todo> search(@NonNull String text) {
        return repository.search(text);
    }
}

File: php/src/ToDo/TodoService.php
<?php

namespace Com\CleanTests\ToDo;

class TodoService
{
    private object $todoRepository;

    public function __construct($todoRepository)
    {
        $this->todoRepository = $todoRepository;
    }

    public function search(string $text): array
    {
        return $this->todoRepository->search($text);
    }
}